nexus/test-utils/src/nexus_test.rs

```diff
@@ -38,6 +38,100 @@ use std::collections::HashMap;
 use std::sync::{Arc, RwLock};
 use std::time::Duration;
 
+pub struct ControlPlaneBuilder<'a> {
+    // required
+    test_name: &'a str,
+
+    // fields with defaults that are cheap to construct
+    nextra_sled_agents: u16,
+    sim_mode: sim::SimMode,
+    tls_cert: Option<Certificate>,
+    gateway_config_file: Utf8PathBuf,
+
+    // fields with defaults that need to be constructed lazily
+    nexus_config: Option<NexusConfig>,
+}
+
+impl<'a> ControlPlaneBuilder<'a> {
+    pub fn new(test_name: &'a str) -> ControlPlaneBuilder<'a> {
+        ControlPlaneBuilder {
+            test_name,
+            nextra_sled_agents: 0,
+            sim_mode: sim::SimMode::Explicit,
+            tls_cert: None,
+            gateway_config_file: DEFAULT_SP_SIM_CONFIG.into(),
+            nexus_config: None,
+        }
+    }
+
+    pub fn extra_sled_agents(mut self, nextra: u16) -> ControlPlaneBuilder<'a> {
+        self.nextra_sled_agents = nextra;
+        self
+    }
+
+    pub fn sim_mode(
+        mut self,
+        sim_mode: sim::SimMode,
+    ) -> ControlPlaneBuilder<'a> {
+        self.sim_mode = sim_mode;
+        self
+    }
+
+    pub fn tls_cert(
+        mut self,
+        tls_cert: Option<Certificate>,
+    ) -> ControlPlaneBuilder<'a> {
+        self.tls_cert = tls_cert;
+        self
+    }
+
+    pub fn mgs_config_file(
+        mut self,
+        path: Utf8PathBuf,
+    ) -> ControlPlaneBuilder<'a> {
+        self.gateway_config_file = path;
+        self
+    }
+
+    pub fn nexus_config(
+        mut self,
+        config: NexusConfig,
+    ) -> ControlPlaneBuilder<'a> {
+        self.nexus_config = Some(config);
+        self
+    }
```

Comment:
could we prefix all of these with `with_`? Generally bare method names indicate getters, not setters (we've had issues with these in the past with `SledBuilder` I think)